velox/functions/lib/string/StringImpl.h

```diff
@@ -162,6 +162,95 @@ FOLLY_ALWAYS_INLINE int32_t charToCodePoint(const T& inputString) {
   return codePoint;
 }
 
+template <typename TInString>
+FOLLY_ALWAYS_INLINE int64_t
+hammingDistance(const TInString& left, const TInString& right) {
+  int64_t distance = 0;
+  int64_t leftPosition = 0;
+  int64_t rightPosition = 0;
+  std::string_view leftSV = std::string_view(left.data(), left.size());
+  std::string_view rightSV = std::string_view(right.data(), right.size());
+
+  while (leftPosition < leftSV.size() && rightPosition < rightSV.size()) {
+    int size = 0;
+    auto codePointLeft = utf8proc_codepoint(&leftSV.at(leftPosition), size);
+    leftPosition += size;
```

Comment:
You need a check for a non negative value here, utf8proc_codepoint can return negative values for non utf8 chars (bad utf8 strings) , in which case this will go into an infinite loop.